### DISCUSSION TITLE
Remote subnet becomes unreachable causing traffic loss

### DISCUSSION THREAD

**Discussion by aleemgsl** [strongswan/strongswan#1473] [Q&A]
Hi All, 

We have an ipsec tunnel up between two machine. Below is the configuration of client (ipsec.conf). 

conn "ikev2"
        auto=start
        keyexchange=ikev2
        ike=aes256-sha2_256-modp2048
        esp=aes256-sha2_256-modp2048
        type=tunnel
        leftcert=test.pem
        leftfirewall=yes
        leftsourceip=%modeconfig
        leftsubnet=192.168.84.0/24
        right=x.x.x.x
        rightid=%any
        rightsubnet=y.y.y.0/25
        authby=rsasig
        dpddelay=5
        dpdtimeout=20
        dpdaction=restart
        lifetime=30m
        ikelifetime=24h
        closeaction=restart
        keyingtries=%forever
        fragmentation = accept
        rekey = yes
	reauth=no	

ipsec statusall :

       ikev2[2]: ESTABLISHED 38 minutes ago, 10.73.73.112[C=XX, O=XX, CN=XX-XX-XX-XX]...A.A.A.A[XX-XX-XX-XX]
       ikev2[2]: IKEv2 SPIs: 829b7930dd72298b_i* 1b6fe4cac789a81a_r, public key reauthentication in 23 hours
       ikev2[2]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
       ikev2{5}:  INSTALLED, TUNNEL, reqid 2, ESP in UDP SPIs: c0adba05_i 001519a7_o
       ikev2{5}:  AES_CBC_256/HMAC_SHA2_256_128/MODP_2048, 63112 bytes_i (736 pkts, 1s ago), 63244 bytes_o (734 pkts, 0s ago), rekeying in 3 minutes
       ikev2{5}:   192.168.84.0/24 === y.y.y.0/25
	   
	   
We can reach y.y.y.0/25 on remote side from 192.168.84.0/24. But after some time, the ping stops working and then again starts to work without any manual intervention. 

There is no specific pattern or timing for this occurrence. 

Can you please suggest what could be the reason for such behavior and how it can be avoided?

We tried setting rekey = no,  reauth = no, make_before_break =yes.

**Comment by Thermi** [comment]
Please follow #196

**Reply by aleemgsl** [reply]
Hi Thermi,

Please let us know what else do you need. Below is some additional information. 

IPsec version : 

Linux strongSwan U5.8.2/K5.15.0-56-generic
University of Applied Sciences Rapperswil, Switzerland
See 'ipsec --copyright' for copyright information.


IPSec statusall when everything works fine:

Status of IKE charon daemon (strongSwan 5.8.2, Linux 5.15.0-56-generic, x86_64):
  uptime: 2 minutes, since Dec 27 07:54:19 2022
  malloc: sbrk 3084288, mmap 0, used 1148320, free 1935968
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 4
  loaded plugins: charon test-vectors ldap pkcs11 tpm aesni aes rc2 sha2 sha1 md5 mgf1 rdrand random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl gcrypt af-alg fips-prf gmp curve25519 agent chapoly xcbc cmac hmac ctr ccm gcm ntru drbg curl attr kernel-netlink resolve socket-default connmark farp stroke updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs dhcp lookip error-notify certexpire led addrblock unity counters
Listening IP addresses:
  10.73.73.74
  172.17.0.1
  10.52.100.0
  192.168.251.3
Connections:
       ikev2:  %any...x.x.x.x  IKEv2, dpddelay=5s
       ikev2:   local:  [C=US, O=XX, CN=XX-XX-XX-XX] uses public key authentication
       ikev2:    cert:  "C=US, O=XX, CN=XX-XX-XX-XX"
       ikev2:   remote: uses public key authentication
       ikev2:   child:  10.52.100.0/28 === y.y.y.0/25 TUNNEL, dpdaction=restart
Security Associations (1 up, 0 connecting):
       ikev2[1]: ESTABLISHED 2 minutes ago, 10.73.73.74[C=US, O=XX, CN=XX-XX-XX-XX]...x.x.x.x[XX-XX-XX-XX]
       ikev2[1]: IKEv2 SPIs: d7c4d79fbaba7efb_i* b1df5e7387b16b14_r, public key reauthentication in 23 hours
       ikev2[1]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
       ikev2{1}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: c800f5ef_i 0900ed21_o
       ikev2{1}:  AES_CBC_256/HMAC_SHA2_256_128, 18920 bytes_i (208 pkts, 1s ago), 18964 bytes_o (209 pkts, 1s ago), rekeying in 10 minutes
       ikev2{1}:   192.168.84.0/28 === y.y.y.0/25


IPSec statusall when issue occurs:

root@GS-4089:/home/ubuntu# ipsec statusall
Status of IKE charon daemon (strongSwan 5.8.2, Linux 5.15.0-56-generic, x86_64):
  uptime: 11 hours, since Dec 26 20:20:51 2022
  malloc: sbrk 3096576, mmap 0, used 1416592, free 1679984
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 5
  loaded plugins: charon test-vectors ldap pkcs11 tpm aesni aes rc2 sha2 sha1 md5 mgf1 rdrand random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl gcrypt af-alg fips-prf gmp curve25519 agent chapoly xcbc cmac hmac ctr ccm gcm ntru drbg curl attr kernel-netlink resolve socket-default connmark farp stroke updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls eap-peap eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs dhcp lookip error-notify certexpire led addrblock unity counters
Listening IP addresses:
  10.73.73.74
  172.17.0.1
  10.52.100.0
  192.168.251.3
Connections:
       ikev2:  %any...x.x.x.x  IKEv2, dpddelay=5s
       ikev2:   local:  [C=US, O=XX, CN=XX-XX-XX-XX] uses public key authentication
       ikev2:    cert:  "C=US, O=XX, CN=XX-XX-XX-XX"
       ikev2:   remote: uses public key authentication
       ikev2:   child:  192.168.84.0/28 === y.y.y.0/25 TUNNEL, dpdaction=restart
Security Associations (1 up, 0 connecting):
       ikev2[1]: ESTABLISHED 11 hours ago, 10.73.73.74[C=US, O=XX, CN=XX-XX-XX-XX]...x.x.x.x[XX-XX-XX-XX]
       ikev2[1]: IKEv2 SPIs: 16463c0192a54cd5_i* a03bf42b29e3d541_r, public key reauthentication in 12 hours
       ikev2[1]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048



In /var/log/syslog below logs can be seen when remote site is not reachable.  There are no additional errors printed in syslogs.

Dec 27 07:35:08 GS-4089 charon-custom: 09[IKE] sending DPD request
Dec 27 07:35:08 GS-4089 charon-custom: 09[ENC] generating INFORMATIONAL request 7759 [ ]
Dec 27 07:35:08 GS-4089 charon-custom: 09[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)
Dec 27 07:35:08 GS-4089 charon-custom: 10[NET] received packet: from x.x.x.x[4500] to 10.73.73.74[4500] (112 bytes)
Dec 27 07:35:08 GS-4089 charon-custom: 10[ENC] parsed INFORMATIONAL response 7759 [ ]
Dec 27 07:35:13 GS-4089 charon-custom: 09[IKE] sending DPD request
Dec 27 07:35:13 GS-4089 charon-custom: 09[ENC] generating INFORMATIONAL request 7760 [ ]
Dec 27 07:35:13 GS-4089 charon-custom: 09[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)
Dec 27 07:35:13 GS-4089 charon-custom: 02[NET] received packet: from x.x.x.x[4500] to 10.73.73.74[4500] (160 bytes)
Dec 27 07:35:13 GS-4089 charon-custom: 02[ENC] parsed INFORMATIONAL response 7760 [ ]





Ipsec.conf

config setup
        #charondebug="ike 3, knl 3, cfg 3, chd 3, dmn 3"
        charondebug="dmn 2, mgr 2, ike 2, chd 2, job 2, enc 4, knl 2, enc 5, net 2, asn 2, lib 5, esp 5, tls 2, tnc 2, imc 2, imv 2, pts 2, cfg 5"
conn "ikev2"
auto=start
keyexchange=ikev2
ike=aes256-sha2_256-modp2048
esp=aes256-sha2_256-modp2048
type=tunnel
leftcert=test.pem
leftfirewall=yes
leftsourceip=%modeconfig
leftsubnet=192.168.84.0/24
right=x.x.x.x
rightid=%any
rightsubnet=y.y.y.0/25
authby=rsasig
dpddelay=5
dpdtimeout=20
dpdaction=restart
lifetime=30m
ikelifetime=24h
closeaction=restart
keyingtries=%forever
fragmentation = accept
rekey = yes

**Reply by aleemgsl** [reply]
Adding more logs : 


		
Dec 27 09:37:57 GS-4089 charon-custom: 10[IKE] sending DPD request
Dec 27 09:37:57 GS-4089 charon-custom: 10[ENC] generating INFORMATIONAL request 16 [ ]
Dec 27 09:37:57 GS-4089 charon-custom: 10[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)
Dec 27 09:38:01 GS-4089 charon-custom: 12[IKE] retransmit 1 of request with message ID 16
Dec 27 09:38:01 GS-4089 charon-custom: 12[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)



Dec 27 09:38:08 GS-4089 charon-custom: 10[IKE] retransmit 2 of request with message ID 16
Dec 27 09:38:08 GS-4089 charon-custom: 10[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)


Dec 27 09:38:21 GS-4089 charon-custom: 15[IKE] retransmit 3 of request with message ID 16
Dec 27 09:38:21 GS-4089 charon-custom: 15[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)


Dec 27 09:38:45 GS-4089 charon-custom: 15[IKE] retransmit 4 of request with message ID 16
Dec 27 09:38:45 GS-4089 charon-custom: 15[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (80 bytes)


Dec 27 09:40:42 GS-4089 charon-custom: 02[IKE] giving up after 5 retransmits
Dec 27 09:40:42 GS-4089 charon-custom: 02[IKE] installing new virtual IP z.z.z.z
Dec 27 09:40:42 GS-4089 charon-custom: 02[IKE] restarting CHILD_SA ikev2
Dec 27 09:40:42 GS-4089 charon-custom: 02[IKE] initiating IKE_SA ikev2[2] to x.x.x.x
Dec 27 09:40:42 GS-4089 charon-custom: 02[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Dec 27 09:40:42 GS-4089 charon-custom: 02[NET] sending packet: from 10.73.73.74[500] to x.x.x.x[500] (1364 bytes)
Dec 27 09:40:42 GS-4089 vpn: - XX-XX-XX-XX y.y.y.0/25 == x.x.x.x -- 10.73.73.74 == y.y.y.0/28
Dec 27 09:40:43 GS-4089 charon-custom: 13[NET] received packet: from x.x.x.x[500] to 10.73.73.74[500] (437 bytes)
Dec 27 09:40:43 GS-4089 charon-custom: 13[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) CERTREQ ]
Dec 27 09:40:43 GS-4089 charon-custom: 13[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Dec 27 09:40:43 GS-4089 charon-custom: 13[IKE] local host is behind NAT, sending keep alives
Dec 27 09:40:43 GS-4089 charon-custom: 13[IKE] remote host is behind NAT
Dec 27 09:40:43 GS-4089 charon-custom: 13[IKE] authentication of 'C=US, O=XX, CN=XX-XX-XX-XX' (myself) with RSA signature successful
Dec 27 09:40:43 GS-4089 charon-custom: 13[IKE] sending end entity cert "C=US, O=XX, CN=XX-XX-XX-XX"
Dec 27 09:40:43 GS-4089 charon-custom: 13[IKE] establishing CHILD_SA ikev2{9} reqid 1
Dec 27 09:40:43 GS-4089 charon-custom: 13[ENC] generating IKE_AUTH request 1 [ IDi CERT AUTH CPRQ(ADDR DNS) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
Dec 27 09:40:43 GS-4089 charon-custom: 13[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (1776 bytes)

After this there is no answer from server. 


Client retransmit request for 5 more times and then inititaes new connection. This time it is getting successful.


Dec 27 09:46:13 GS-4089 charon-custom: 14[IKE] giving up after 5 retransmits
Dec 27 09:46:13 GS-4089 charon-custom: 14[IKE] peer not responding, trying again (3/0)
Dec 27 09:46:13 GS-4089 charon-custom: 14[IKE] initiating IKE_SA ikev2[2] to x.x.x.x
Dec 27 09:46:13 GS-4089 charon-custom: 14[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Dec 27 09:46:13 GS-4089 charon-custom: 14[NET] sending packet: from 10.73.73.74[500] to x.x.x.x[500] (1364 bytes)
Dec 27 09:46:14 GS-4089 charon-custom: 02[NET] received packet: from x.x.x.x[500] to 10.73.73.74[500] (437 bytes)
Dec 27 09:46:14 GS-4089 charon-custom: 02[ENC] parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) CERTREQ ]
Dec 27 09:46:14 GS-4089 charon-custom: 02[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Dec 27 09:46:14 GS-4089 charon-custom: 02[IKE] authentication of 'C=US, O=XX, CN=XX-XX-XX-XX' (myself) with RSA signature successful
Dec 27 09:46:14 GS-4089 charon-custom: 02[IKE] sending end entity cert "C=US, O=XX, CN=XX-XX-XX-XX"
Dec 27 09:46:14 GS-4089 charon-custom: 02[IKE] establishing CHILD_SA ikev2{11}
Dec 27 09:46:14 GS-4089 charon-custom: 02[ENC] generating IKE_AUTH request 1 [ IDi CERT AUTH CPRQ(ADDR DNS) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) N(ADD_4_ADDR) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
Dec 27 09:46:14 GS-4089 charon-custom: 02[NET] sending packet: from 10.73.73.74[4500] to x.x.x.x[4500] (1776 bytes)
Dec 27 09:46:14 GS-4089 charon-custom: 12[NET] received packet: from x.x.x.x[4500] to 10.73.73.74[4500] (1252 bytes)
Dec 27 09:46:14 GS-4089 charon-custom: 12[ENC] parsed IKE_AUTH response 1 [ EF(1/2) ]
Dec 27 09:46:14 GS-4089 charon-custom: 12[ENC] received fragment #1 of 2, waiting for complete IKE message
Dec 27 09:46:14 GS-4089 charon-custom: 11[NET] received packet: from x.x.x.x[4500] to 10.73.73.74[4500] (820 bytes)
Dec 27 09:46:14 GS-4089 charon-custom: 11[ENC] parsed IKE_AUTH response 1 [ EF(2/2) ]
Dec 27 09:46:14 GS-4089 charon-custom: 11[ENC] received fragment #2 of 2, reassembled fragmented IKE message (1584 bytes)
Dec 27 09:46:14 GS-4089 charon-custom: 11[ENC] parsed IKE_AUTH response 1 [ CERT IDr AUTH CPRP(ADDR MASK) TSi TSr SA ]
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] received end entity cert "C=US, O=XX, CN=XX-XX-XX-XX"
Dec 27 09:46:14 GS-4089 charon-custom: 11[CFG] no issuer certificate found for "C=US, O=XX, CN=XX-XX-XX-XX"
Dec 27 09:46:14 GS-4089 charon-custom: 11[CFG]   issuer is "C=US, O=XX, CN=XX Root CA"
Dec 27 09:46:14 GS-4089 charon-custom: 11[CFG]   reached end of incomplete trust chain for trusted certificate "C=US, O=XX, CN=XX-XX-XX-XX"
Dec 27 09:46:14 GS-4089 charon-custom: 11[CFG]   using trusted certificate "C=US, O=XX, CN=XX-XX-XX-XX"
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] authentication of 'XX-XX-XX-XX' with RSA signature successful
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] IKE_SA ikev2[2] established between 10.73.73.74[C=US, O=XX, CN=XX-XX-XX-XX]...x.x.x.x[XX-XX-XX-XX]
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] scheduling reauthentication in 85700s
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] maximum IKE_SA lifetime 86240s
Dec 27 09:46:14 GS-4089 charon-custom: 11[CFG] handling INTERNAL_IP4_NETMASK attribute failed
Dec 27 09:46:14 GS-4089 avahi-daemon[715]: Withdrawing address record for z.z.z.z on enp0s31f6.
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] installing new virtual IP z.z.z.z
Dec 27 09:46:14 GS-4089 avahi-daemon[715]: Registering new address record for z.z.z.z on enp0s31f6.IPv4.
Dec 27 09:46:14 GS-4089 charon-custom: 11[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ
Dec 27 09:46:14 GS-4089 charon-custom: 11[IKE] CHILD_SA ikev2{11} established with SPIs ca54664a_i 0c6f6fff_o and TS 10.52.100.0/28 === y.y.y.0/25
Dec 27 09:46:14 GS-4089 vpn: + XX-XX-XX-XX y.y.y.0/25 == x.x.x.x -- 10.73.73.74 == y.y.y.0/28

**Reply by aleemgsl** [reply]
Hi, 

Can you please suggest if this frequent disconnect can be avoided by any config changes?
